### PR TITLE
OUT-581 Task does not have the right code

### PR DESCRIPTION
**Pull request by aatbip**
### Changes

- [x] Implements delete during update task (previously unassigned) and delete task

### CODE REVIEW

File: src/app/api/label-mapping/label-mapping.service.ts [hunk 1]
```diff
@@ -163,4 +163,17 @@ export class LabelMappingService extends BaseService {
       },
     })
   }
+
+  async deleteLabel(label: string) {
+    const currentLabel = await this.db.label.findFirst({
+      where: {
+        label: label,
+      },
+    })
```
Comment (rrojan) [inline src/app/api/label-mapping/label-mapping.service.ts:172]:
```suggestion
    const currentLabel = await this.db.label.findFirst({ where: { label } })
```

File: src/app/api/tasks/tasks.service.ts [hunk 2]
```diff
@@ -220,6 +222,9 @@ export class TasksService extends BaseService {
       const notificationsService = new NotificationService(this.user)
       await notificationsService.markClientNotificationAsRead(task)
     }
+    //delete the associated label
+    const labelMappingService = new LabelMappingService(this.user)
+    await labelMappingService.deleteLabel(task?.label as string)
```
Comment (rrojan) [inline src/app/api/tasks/tasks.service.ts:227]: Why is task possibly undefined here? Is there a condition where this is possible and has it been accounted for?

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/app/api/tasks/tasks.service.ts (hunk 2 above)
Comment (aatbip) [reply src/app/api/tasks/tasks.service.ts]: changed made in the commit below

File: src/app/api/label-mapping/label-mapping.service.ts (hunk 1 above)
Comment (aatbip) [reply src/app/api/label-mapping/label-mapping.service.ts]: changes had been applied in the commit below

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Pull request merged**